Move exception to main thread in ParallelForestTraversalExecutor

There are two places inside ParallelForestTraversalExecutor where a lambda being executed on a threadpool can throw an exception. We should move this exception outside the lambda to the main execution thread. This should give more predictable behavior and a better stack trace.

* https://github.com/aws/random-cut-forest-by-aws/blob/master/src/main/java/com/amazon/randomcutforest/ParallelForestTraversalExecutor.java#L65
* https://github.com/aws/random-cut-forest-by-aws/blob/master/src/main/java/com/amazon/randomcutforest/ParallelForestTraversalExecutor.java#L122

The current code, which throws the exception inside the threadpool task, does not compile on AL 2012.

```
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /local/home/joshtok/workplace/random-cut-forest-by-aws/src/main/java/com/amazon/randomcutforest/ParallelForestTraversalExecutor.java:[65,25] unreported exception X; must be caught or declared to be thrown
[ERROR] /local/home/joshtok/workplace/random-cut-forest-by-aws/src/main/java/com/amazon/randomcutforest/ParallelForestTraversalExecutor.java:[122,25] unreported exception X; must be caught or declared to be thrown
```

--- a/src/main/java/com/amazon/randomcutforest/ParallelForestTraversalExecutor.java
+++ b/src/main/java/com/amazon/randomcutforest/ParallelForestTraversalExecutor.java
@@ -62,8 +62,7 @@ public class ParallelForestTraversalExecutor extends AbstractForestTraversalExec
             })
             .reduce(accumulator)
             .map(finisher)
-            .orElseThrow(() -> new IllegalStateException("accumulator returned an empty result"))
-        );
+        ).orElseThrow(() -> new IllegalStateException("accumulator returned an empty result"));
     }
 
     @Override
@@ -119,8 +118,7 @@ public class ParallelForestTraversalExecutor extends AbstractForestTraversalExec
             })
             .reduce(accumulator)
             .map(finisher)
-            .orElseThrow(() -> new IllegalStateException("accumulator returned an empty result"))
-        );
+        ).orElseThrow(() -> new IllegalStateException("accumulator returned an empty result"));
     }
 
     @Override

